File: Exemplos M10/App_IntentDigitalHUB_Flutter/Elgin_IntentDigitalHUB_Flutter-font/lib/pages/balanca/balance_model.dart
// ignore_for_file: constant_identifier_names

part of 'balance_page.dart';

enum BalanceModel { DP3005, SA110, DPSC, DP30CK }

extension BalanceModelExtension on BalanceModel {
  //Retorna o código de modelo correspondente de cada balança
  int get balanceCode {
    switch (this) {
      case BalanceModel.DP3005:
        return 0;
      case BalanceModel.SA110:
        return 1;
      case BalanceModel.DPSC:
        return 2;
      case BalanceModel.DP30CK:
        return 3;
    }
  }
}

File: Exemplos M10/App_IntentDigitalHUB_XamarinForms/Elgin_IntentDigitalHUB_XamarinForms-font/E1_Digital_Hub_Xamarin_Forms_M8/Balance/BalanceModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Xamarin_Forms_Intent_Digital_Hub.Balance
{
    class BalanceModel
    {
        private BalanceModel(int balanceCode) { this.balanceCode = balanceCode; }

        private int balanceCode;
        public int getBalanceCode() { return balanceCode; }

        public static BalanceModel DP3005 = new BalanceModel(0);
        public static BalanceModel SA110 = new BalanceModel(1);
        public static BalanceModel DPSC = new BalanceModel(2);
        public static BalanceModel DP30CK = new BalanceModel(3);
    }
}
